Extend the Look At component's capabilities

gralco opened the pull request:
- Add a strength parameter that determines how quickly the rotation approaches the target, and therefore also how much it resists the change in rotation.
- Add options to disable rotating about X, Y, and/or Z (pitch, roll, and yaw).
- Add a bool that enables or disables the component.
- Add a bool that sets whether the Look At transform is applied (still calculated and retrievable when not applied).
- Add getters for each of the previously existing attributes and add both getters and setters for the new ones.

This was tested with the First Person Controller gem.

<img width="447" height="237" alt="image" src="https://github.com/user-attachments/assets/ec8d05a9-5ecd-4137-a22b-223b108e6863" />
<br>
<img width="248" height="526" alt="image" src="https://github.com/user-attachments/assets/6b8fecec-ead9-4486-a0a0-5725cff84705" />

gralco committed:
Extend the Look At component's capabilities

- Add a strength parameter that determines how quickly the rotation
  approaches the target, and therefore also how much it resists the
  change in rotation.
- Add options to disable rotating about X, Y, and/or Z (pitch, roll,
  and yaw).
- Add a bool that enables or disables the component.
- Add getters for each of the previously existing attributes and add
  both getters and setters for the new ones.

Signed-off-by: Eric Kuzmenko <eric.n.kuzmenko@gmail.com>

gralco commented:
I've tested this on 2510.1 with the First Person Controller gem.

nick-l-o3de commented on Gems/LmbrCentral/Code/Source/Scripting/EditorLookAtComponent.cpp:
```diff
@@ -151,6 +178,13 @@ namespace LmbrCentral
 
+    //=========================================================================
+    void EditorLookAtComponent::OnEnabledChanged()
+    {
+        if (m_enabled)
+            RecalculateTransform();
```
O3DE coding standards requires `{ }` around flow control blocks

https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md

gralco replied:
@nick-l-o3de I've addressed this, thanks.

nick-l-o3de commented on Gems/LmbrCentral/Code/Source/Scripting/EditorLookAtComponent.cpp:
```diff
@@ -161,6 +195,9 @@ namespace LmbrCentral
     //=========================================================================
     void EditorLookAtComponent::RecalculateTransform()
     {
+        if (!m_enabled)
+            return;
+
```
same as above

gralco replied:
@nick-l-o3de I've addressed this, thanks.

nick-l-o3de commented on Gems/LmbrCentral/Code/Source/Scripting/LookAtComponent.h:
```diff
@@ -113,8 +166,12 @@ namespace LmbrCentral
+        bool m_fixatePitch;
+        bool m_fixateRoll;
+        bool m_fixateYaw;
+        bool m_enabled;
```
Should we initialize these to the same values as the editor?

gralco replied:
@nick-l-o3de I've addressed this, thanks.

gralco committed:
Look At: Make the strength application framerate independent, normalized to 30FPS

- Add initial values to LookAtComponent's member variables.
- Wrap flow control blocks with curly braces.

Signed-off-by: Eric Kuzmenko <eric.n.kuzmenko@gmail.com>

gralco committed:
Look At: Add the option to get the Look At transform which can be calculated without applying it

Signed-off-by: Eric Kuzmenko <eric.n.kuzmenko@gmail.com>

gralco committed:
EditorLookAtComponent: Wrap flow control blocks in curly brackets

Signed-off-by: Eric Kuzmenko <eric.n.kuzmenko@gmail.com>

gralco committed:
LookAtComponent: Broadcast OnEnabledChanged when it actually changes in the setter

Signed-off-by: Eric Kuzmenko <eric.n.kuzmenko@gmail.com>

nick-l-o3de commented on Gems/ScriptCanvas/Assets/TranslationAssets/EBus/Senders/LookAt.names:
```diff
@@ -5,30 +5,77 @@
             "context": "EBusSender",
             "variant": "",
             "details": {
-                "name": "Look At",
-                "category": "Gameplay"
```
I don't realy know what a .names file is for, but it appears that this change, it removes the "Category" tag.  Is that intentional?

gralco committed:
Look At: Add the option to either disable the component entirely or to calculate the Look At transform but not apply it

- Bring back the "Gameplay" category in LookAt.names

Signed-off-by: Eric Kuzmenko <eric.n.kuzmenko@gmail.com>

gralco committed:
LookAt: Make normalizedFrameRate a constexpr

Signed-off-by: Eric Kuzmenko <eric.n.kuzmenko@gmail.com>
